### ISSUE TITLE
Angular 9: can't resove 'videojs'

### ISSUE THREAD

**Issue by fosteman** [collab-project/videojs-record#480]
## Description
Module is not found on initialization of an Angular project.

## Steps to reproduce
Installed on Angular 9 with custom webpack config project. 

webpack.partial.js
```
const webpack = require('webpack');
const ProvidePlugin = require('webpack/lib/ProvidePlugin');

module.exports = {
	resolve: {
		alias: {
			videojs: 'video.js',
		}
	},
	plugins: [
		new ProvidePlugin({
			videojs: 'video.js/dist/video.cjs.js',
			RecordRTC: 'recordrtc'
		}),
	]
}
```

Successful Angular compilation is expected

But naturally error occurs, interrupting compilation.
```
ERROR in ./node_modules/videojs-record/dist/videojs.record.js
Module not found: Error: Can't resolve 'videojs' in '/Users/fosteman/PRJ/node_modules/videojs-record/dist'
```

## Additional Information
Running `node v12.13.0`, `Mac os Catalina`
`"@types/video.js": "^7.3.9"`
`"video.js": "^7.7.6"`
`"videojs-record": "^4.0.0"`,
`"core-js": "^3.6.4",`
`"webpack": "latest"`

**Comment by thijstriemstra** [comment]
I can only point you to https://collab-project.github.io/videojs-record/#/frameworks/angular?id=configuration

Are you sure your config is executed? Try a non-videojs-record example, only video.js: https://github.com/videojs/video.js/blob/master/docs/guides/angular.md

**Comment by fosteman** [comment]
@thijstriemstra fool-proofed webpack config, same error. 
As I looked into it further, **solution** came to me: 
`-record` module `require`s `video.js` as a dependency. Error occurs on this enquiry between modules. Naturally, I'd move a copy of `video.js/dist/video.js` over to `videojs-record/dist/videojs`, and change the `require` statement paths to `./videojs` instead of `videojs`.

The reason for the error, though, is mysterious, since it requires looking into webpack logs of loading modules. Perhaps `videojs` **alias** is not loaded up in time.

Any ideas, @thijstriemstra ?

**Comment by thijstriemstra** [comment]
Angular gives me a headache (most of these frameworks do). Maybe start at the basis with the webpack guide: https://collab-project.github.io/videojs-record/#/frameworks/webpack And see what's different between webpack configs. Thanks for investigating this!

**Comment by thijstriemstra** [comment]
also, you never mentioned if you tried the [angular guide](https://github.com/videojs/video.js/blob/master/docs/guides/angular.md) or not? Does that work for you?

**Comment by fosteman** [comment]
In the process of checking the guide you're mentioning.
Latest news are that `video.js`, `videojs-record`, and other modules load correctly. It even successfully renders a `<video/>` element.
Then crash occurs if video-instance is initialized with `record` plugin configured
```
// video.js configuration
		this.config = {
			plugins: {
				// videojs-record plugin
				record: {
					audio: true,
					video: true,
					debug: true
				}
```

**Comment by thijstriemstra** [comment]
> Then crash occurs 

What kind of crash? Screenshot/log?

**Comment by fosteman** [comment]
Regarding the 'How to load `video.js@7.7.6` and `videojs-record@4.0.0`, 

`node_modules/videojs-record/` - Module folder
<img width="190" alt="Screen Shot 2020-05-25 at 11 17 29 AM" src="https://user-images.githubusercontent.com/32850517/82825483-53a86e00-9e79-11ea-8092-8cbe7ed77140.png">

`node_modules/videojs-record/dist/videojs-record.js` looks like this
<img width="811" alt="Screen Shot 2020-05-25 at 11 14 55 AM" src="https://user-images.githubusercontent.com/32850517/82825449-43908e80-9e79-11ea-95b0-3895ec2b30f9.png">

Component imports ```
import videojs from 'video.js';
import * as RecordRTC from 'recordrtc';
import * as Record from 'videojs-record'
```

In this way, `video.js` instantiates and renders successfully, but it refuses to load `videojs-record` plugin if specified in `options` passed in constructor.

**Comment by fosteman** [comment]
![Uploading Screen Shot 2020-05-25 at 11.20.48 AM.png…]()

**Comment by fosteman** [comment]
> In the process of checking the guide you're mentioning.
> Latest news are that `video.js`, `videojs-record`, and other modules load correctly. It even successfully renders a `<video/>` element.
> Then crash occurs if video-instance is initialized with `record` plugin configured
> 
> ```
> // video.js configuration
> 		this.config = {
> 			plugins: {
> 				// videojs-record plugin
> 				record: {
> 					audio: true,
> 					video: true,
> 					debug: true
> 				}
> ```

<img width="662" alt="Screen Shot 2020-05-25 at 11 21 25 AM" src="https://user-images.githubusercontent.com/32850517/82825793-dfba9580-9e79-11ea-9174-586307cafa17.png">

**Comment by thijstriemstra** [comment]
The angular guide provides a working setup, you should compare it to your angular project and spot the webpack config differences. There is no need to modify the videojs-record.js source file and change the imports. The webpack config is there so you don't have to modify the source and do the lib routing in webpack by using the ProvidePlugin, resolve, alias etc.

**Comment by fosteman** [comment]
Following these guidelines end up with this
<img width="587" alt="Screen Shot 2020-05-25 at 4 32 51 PM" src="https://user-images.githubusercontent.com/32850517/82841804-633dac00-9ea5-11ea-9fbe-92c4301f31eb.png">

**Comment by thijstriemstra** [comment]
I tested the angular guide just now and everything worked fine: https://collab-project.github.io/videojs-record/#/frameworks/angular?id=configuration

**Comment by dineshmaths1** [comment]
> Following these guidelines end up with this
> <img alt="Screen Shot 2020-05-25 at 4 32 51 PM" width="587" src="https://user-images.githubusercontent.com/32850517/82841804-633dac00-9ea5-11ea-9fbe-92c4301f31eb.png">

Did you fixed . i am facing same issue

**Comment by fosteman** [comment]
@dineshmaths1 , nope, I didn't follow through the solution to it. As far as I know, the closest person to the problem is https://www.linkedin.com/in/vatsalnshah/ .

**Comment by thijstriemstra** [comment]
Fixed in v4.1.0.
